airbyte-cdk/python/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py

```diff
@@ -86,110 +79,67 @@ def name(self, value: str) -> None:
         if not isinstance(value, property):
             self._name = value
 
-    @property
-    def url_base(self) -> str:
-        return self.requester.get_url_base()
-
-    @property
-    def http_method(self) -> str:
-        return str(self.requester.get_method().value)
-
-    @property
-    def raise_on_http_errors(self) -> bool:
-        # never raise on http_errors because this overrides the error handler logic...
-        return False
-
-    @property
-    def max_retries(self) -> Union[int, None]:
-        if self.disable_retries:
-            return 0
-        # this will be removed once simple_retriever is decoupled from http_stream
-        if hasattr(self.requester.error_handler, "max_retries"):  # type: ignore
-            return self.requester.error_handler.max_retries  # type: ignore
-        return self._DEFAULT_MAX_RETRY
-
-    def should_retry(self, response: requests.Response) -> bool:
-        """
-        Specifies conditions for backoff based on the response from the server.
-
-        By default, back off on the following HTTP response statuses:
-         - 429 (Too Many Requests) indicating rate limiting
-         - 500s to handle transient server errors
-
-        Unexpected but transient exceptions (connection timeout, DNS resolution failed, etc..) are retried by default.
+    def _get_mapping(
```

Comment:
This is duplicated from HttpRequester. Should we extract the logic in a "OptionMerger" class or something? I think the broader question is: if one change, will the other change? Right now, it seems like both will evolve at the same pace and we should therefore extract this

Reply:
I think you are right, I will abstract this out.